File: Python-DAS-ALGO/Selection-Sort/python/Selection-Sort.py
def selection_sort(number, size):

    # outer loop for all numbers
    for i in range(size):
        min_index = i

        # inner loop for find the min index
        for j in range(i+1, size):
            if number[j] < number[min_index]:
                min_index = j

        # swap two numbers
        if min_index != i:
            # first way for swap
            # temp = number[i]
            # number[i] = number[min_index]
            # number[min_index] = temp

            # second way for swap
            (number[i], number[min_index]) = (number[min_index], number[i])


numbers = [3, 44, 38, 1, 27, 2, 4]
size = len(numbers)
selection_sort(numbers, size)
print("After sorting array")
print(numbers)

File: Python-DSA-ALGO/Selection-Sort/js/Selection-Sort.js

const numbers = [3, 44, 38, 1, 27, 2, 4]

// first iteration --> 


// outer loop for all numbers

for(let i = 0; i < numbers.length - 1; i++) {

    let min_index = i

    // inner loop for find the min index
    for(let j = i + 1; j < numbers.length; j++) {
        if(numbers[j] < numbers[min_index]) {
            min_index = j
        }
    }  
    
    // swap two numbers
    if(min_index != i) {

        [numbers[i], numbers[min_index]] = [numbers[min_index], numbers[i]]

        // let temp = numbers[i]
        // numbers[i] = numbers[min_index]
        // numbers[min_index] = temp
    }
}

for(let i of numbers) {
    console.log(i)
}